### PR TITLE
Back merge master to read cache release branch

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
Brings master changes to read cache release branch.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - Ran as part of presubmit
3. Integration tests - Ran with KOKORO (manually verified no failures)
4. Perf tests - 
Screenshot 2024-02-05 at 11 00 45

### CODE REVIEW

File: perfmetrics/scripts/ml_tests/pytorch/run_model.sh [hunk 1]
```diff
@@ -30,6 +31,12 @@ cd -
...
 
+# We have created a bucket in the asia-northeast1 region to align with the location of our PyTorch 2.0 VM, which is also in asia-northeast1.
+if [ ${PYTORCH_VESRION} == "v2" ];
+then
+  TEST_BUCKET="gcsfuse-ml-data-asia-northeast1"
+fi
+
 config_filename=/tmp/gcsfuse_config.yaml
 cat > $config_filename << EOF
 logging:
```
Comment (ashmeenkaur) [inline perfmetrics/scripts/ml_tests/pytorch/run_model.sh:42]: There were conflicts in this file @Tulsishah @gargnitingoogle. PTAL!

Comment (gargnitingoogle) [reply perfmetrics/scripts/ml_tests/pytorch/run_model.sh:42]: Sure. But this change wasn't made by me, so I'm not aware of its correctness.

File: perfmetrics/scripts/ml_tests/pytorch/run_model.sh [hunk 2]
```diff
@@ -46,13 +53,14 @@ EOF
 echo "Created config-file at "$config_filename
```
Comment (Tulsishah) [inline perfmetrics/scripts/ml_tests/pytorch/run_model.sh:53]: Master is different in adding metadata-cache-ttl sec 
https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/perfmetrics/scripts/ml_tests/pytorch/run_model.sh

My changes are okay of changing bucket name and removing epochs.
https://github.com/GoogleCloudPlatform/gcsfuse/pull/1610

File: perfmetrics/scripts/ml_tests/pytorch/run_model.sh (hunk 1 above)
Comment (ashmeenkaur) [reply perfmetrics/scripts/ml_tests/pytorch/run_model.sh:42]: One of the source of conflicts was https://github.com/GoogleCloudPlatform/gcsfuse/commit/71840f520a8677056118a8d501da84d6cdb69fd4

File: perfmetrics/scripts/ml_tests/pytorch/run_model.sh (hunk 2 above)
Comment (ashmeenkaur) [reply perfmetrics/scripts/ml_tests/pytorch/run_model.sh:53]: Yes this is where I was unsure. @gargnitingoogle, can you confirm if type-cache-ttl, stat-cache-ttl flags should be removed? Can you also check if any other changes are required?

Comment (gargnitingoogle) [reply perfmetrics/scripts/ml_tests/pytorch/run_model.sh:53]: > can you confirm if type-cache-ttl, stat-cache-ttl flags should be removed?

Yes, we should not use type-cache-ttl or stat-type-ttl in the read_cache_release branch now as they are to be deprecated and replaced with ttl-secs config. So, the change on line 56-57 (on RHS) should be reverted.

> Can you also check if any other changes are required?

All other changes of mine look fine to me.

**Review by gargnitingoogle — APPROVED** [review]
LGTM for my changes

**Review by sethiay — COMMENTED** [review]
The changes are mostly from @Tulsishah so would be great if @Tulsishah can give approval.

**Review by Tulsishah — APPROVED** [review]

**Review by Tulsishah — APPROVED** [review]

**Review by raj-prince — APPROVED** [review]

**Pull request merged**